packages/gatsby-transformer-sharp/src/customize-schema.js

```diff
@@ -441,10 +442,16 @@ const createFields = ({
             // this is no longer in progress
             inProgressCopy.delete(publicPath)
             if (err) {
-              console.error(
-                `error copying file from ${details.absolutePath} to ${publicPath}`,
-                err
-              )
+              // Capture error within new Reporter API
+              if (reporter.setErrorMap) {
```

Comment:
Same suggestion here. We can call it in a backwards compatible way by putting the message in `context.sourceMessage`, and then we can actually pass the `err` object as the second argument to `panic`